### PR TITLE
MM-43277: fix channel actions modal dropdown cutoffs/overflows

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary
`RunPlaybookChildren` & `CategorySelector` use `react-select` directly. It's built-in portal behavior isn't active by default. Our `GenericModal` is set at zIndex: 1040, and portals directly to `document.body` by default.

- fix: Channel Actions modal dropdowns—portal and zIndex to 1041 (https://github.com/mattermost/mattermost-plugin-playbooks/commit/a3f6cf590f9cf488bc55cea0cd9e5687b32b3b93)

extra:
- aside: drop internal `Portal` and use `FloatingPortal` (https://github.com/mattermost/mattermost-plugin-playbooks/commit/fbe17969ecf4ca92ec9a20784b8a8b3e7c13604e, https://github.com/mattermost/mattermost-plugin-playbooks/commit/3a729183d9c7675e47506e6d8893bb0c4d0aa2c9)
affected: checklist & checklist_item drag&drop, but nothing has changed
- aside: use auto update functionality in `DotMenu`—e.g. flip on scrolling when near viewport edge (https://github.com/mattermost/mattermost-plugin-playbooks/commit/2ed38965842a52a99afcdd1f10068678f4651c7f)

#### Screenshots
| before | after |
|---|---|
| - | image  | 
| - | image |

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-43277

#### Checklist

Item | Y / N / NA [^1]
---|---
Telemetry updated? | N
Gated by experimental feature flag? | N
Unit/E2E tests updated? | https://github.com/mattermost/mattermost-plugin-playbooks/pull/1418/commits/141254c36e7d64f4bba12ac4f6961ab53e651fb8, https://github.com/mattermost/mattermost-plugin-playbooks/pull/1418/commits/a2a81a9dd8009a00e2303d4ccf0d069d361d9da1, https://github.com/mattermost/mattermost-plugin-playbooks/pull/1418/commits/18c0cf004e668a7a6ca43813a520b5decd8a76ff

[^1]: Yes / No / Not Applicable.

### CODE REVIEW

File: webapp/src/components/actions_modal_action_children.tsx [hunk 1]
```diff
@@ -73,8 +76,10 @@ export const RunPlaybookChildren = ({playbookId, onUpdate, editable}: RunPlayboo
             value={playbookOptions?.find((p) => p.id === playbookId)}
             isClearable={false}
             maxMenuHeight={250}
-            styles={{indicatorSeparator: () => null}}
+            styles={{indicatorSeparator: () => null, menuPortal: (base: CSSProperties) => ({...base, zIndex: 1041})}}
```
Comment (lieut-data) [inline webapp/src/components/actions_modal_action_children.tsx:79]: Is it feasible to import `1040+1` as a constant from the webapp?

**Review by lieut-data — APPROVED** [review]
Sweet!

File: webapp/src/components/checklist/checklist_list.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ import {
 
 import classNames from 'classnames';
 
-import Portal from 'src/components/portal';
+import {FloatingPortal} from '@floating-ui/react-dom-interactions';
```
Comment (iomodo) [inline webapp/src/components/checklist/checklist_list.tsx:20]: nice!

**Review by iomodo — APPROVED** [review]
Looks great @calebroseland !

**Comment by calebroseland** [comment]
Alright, so a little bit of retooling here as react-select's built in portal functionality causes a few issues with jump-glitches and unanchored positioning on scroll:
CleanShot 2022-08-17 at 16 04 04


Fixed by removing the portaling and letting react-select just open up or down automatically, and then using the overflow: visible approach instead of inner scrollbars. (This makes the outer modal scroll on short screens)



CleanShot 2022-08-17 at 16 06 03



Also included is a fix to make all `GenericModal`s in Playbooks to work on short and narrow screens. (No more negative margin hacks!) 2/3s of freespace goes below, and 1/3 goes above the modal to float in a nice position at all viewport heights.

CleanShot 2022-08-17 at 16 07 52
CleanShot 2022-08-17 at 16 11 46

**Review by lieut-data — APPROVED** [review]

**Review by iomodo — APPROVED** [review]

**Pull request merged**